src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/StartAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,164 @@
+            ParameterSetName = "SetByName")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Location of the network watcher.",
+            ParameterSetName = "SetByLocation")]
+        [LocationCompleter("Microsoft.Network/networkWatchers/connectionMonitors")]
+        [ValidateNotNull]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource ID.",
+            ParameterSetName = "SetByResourceId")]
+        [ValidateNotNull]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "Connection monitor object.",
+            ParameterSetName = "SetByInputObject")]
+        [ValidateNotNull]
+        public PSConnectionMonitorResult InputObject { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocation")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            string connectionMonitorName = this.Name;
+            string resourceGroupName = string.Empty;
+            string networkWatcherName = string.Empty;
+
+            if (ParameterSetName.Contains("SetByResourceId"))
+            {
+                ConnectionMonitorDetails connectionMonitorDetails = new ConnectionMonitorDetails();
+                connectionMonitorDetails = this.GetConnectionMonitorDetails(this.ResourceId);
+
+                connectionMonitorName = connectionMonitorDetails.ConnectionMonitorName;
+                resourceGroupName = connectionMonitorDetails.ResourceGroupName;
+                networkWatcherName = connectionMonitorDetails.NetworkWatcherName;
+            }
+            else if (ParameterSetName.Contains("SetByResource"))
+            {
+                resourceGroupName = this.NetworkWatcher.ResourceGroupName;
+                networkWatcherName = this.NetworkWatcher.Name;
+            }
+            else if (ParameterSetName.Contains("SetByName"))
```

Comment:
@irrogozh same comment about removing this block and setting the above variables to the respective parameter values

Reply:
Done